mir_eval/chord.py

```diff
@@ -1342,6 +1347,136 @@ def sevenths_inv(reference_labels, estimated_labels):
     return comparison_scores
 
 
+def directional_hamming_distance(reference_intervals, estimated_intervals):
+    """Compute the directional hamming distance between reference and
+    estimated intervals as defined by [#harte2010towards]_ and used for MIREX
+    'OverSeg', 'UnderSeg' and 'MeanSeg' measures.
+
+    Examples
+    --------
+    >>> (ref_intervals,
+    ...  ref_labels) = mir_eval.io.load_labeled_intervals('ref.lab')
+    >>> (est_intervals,
+    ...  est_labels) = mir_eval.io.load_labeled_intervals('est.lab')
+    >>> overseg = 1 - mir_eval.chord.directional_hamming_distance(
+    ...     ref_intervals, est_intervals)
+    >>> underseg = 1 - mir_eval.chord.directional_hamming_distance(
+    ...     est_intervals, ref_intervals)
+    >>> meanseg = min(overseg, underseg)
```

Comment:
I'm confused by this line --
- It's called "meanseg" but is computed as the min.
- Is this a thing people measure?  If so should we have a separate metric function for it?

Reply:
I agree that the naming is weird, but I wanted to be consistent with MIREX (see http://www.music-ir.org/mirex/wiki/2017:Audio_Chord_Estimation_Results).

I'm not sure if it is meaningful to have this as a separate metric for this for each song (it is, after all, just min(overseg, underseg)). However, it might then be easier to recreate the metrics as used in the MIREX task, where "MeanSeg" means the mean of the min(os, us) for each song. In this case, I would prefer to call it just something like "Segmentation".

What do you think?

Reply:
So they are calling it "mean" because it's the mean of the min across all songs?  I guess that bothers me because all of the metrics are mean across songs, correct?  In terms of what I think, I honestly don't know in this case!  But I will say typically it's ok to have an extra metric function if it's something that people measure, even if it's oneline.  The idea being that evaluate returns everything that you might want to measure to report your results.

Reply:
Sure, let's add a function for it, then! The only negative now is that `underseg()` and `overseg()` get computed twice when calling `evaluate()`. If this is a problem, I can replace `scores['seg'] = seg(...)` with `scores['seg'] = min(scores['underseg'], scores['overseg'])`.

Reply:
> If this is a problem, I can replace scores['seg'] = seg(...) with scores['seg'] = min(scores['underseg'], scores['overseg']).

This seems like the best option. (and sorry for the delay)

Reply:
No worries! Should be done now. If there is anything I can do to further improve this PR, let me know.